### PR TITLE
Made AppVeyor and TravisCI use similar code

### PR DESCRIPTION
**Pull request by TravisEz13**
redirect mc messages to stderr which occur even on success to stdout
Removed travis-ci branch restriction now that we are using forks ( makes it easier to use your own instance of travis-ci)

Address a portion of #1646 

---

This change is [Reviewable](https://reviewable.io/reviews/powershell/powershell/2128)

### CODE REVIEW

**Comment by TravisEz13** [comment]
Verified AppVeyor fails when tests fails here:  https://ci.appveyor.com/project/TravisEz13/powershell/build/6.0.0-alpha.9-37

**Comment by TravisEz13** [comment]
Verified Travis-CI fails when tests fails here: 
https://travis-ci.org/TravisEz13/PowerShell/builds/156109750

File: build.psm1 [hunk 1]
```diff
@@ -551,30 +551,32 @@ function Start-PSPester {
...
 
-    $powershell = Get-PSOutput
-
     # All concatenated commands/arguments are suffixed with the delimiter (space)
     $Command = ""
+    $powershell = Join-Path $binDir 'powershell'
 
     # Windows needs the execution policy adjusted
     if ($IsWindows) {
         $Command += "Set-ExecutionPolicy -Scope Process Unrestricted; "
```
Comment (JamesWTruher) [inline build.psm1]: >Unrestricted [](start = 56, length = 12)

I wonder if this should be `ByPass`

File: build.psm1 [hunk 2]
```diff
@@ -585,17 +587,49 @@ function Start-PSPester {
...
+            & $powershell -noprofile -c $Command
+        } finally {
+            $env:PSMODULEPATH = $originalModulePath
+        }        
+    }
+    if($ThrowOnFailure)
+    {
+        Test-PSPesterResults -TestResultsFile $OutputFile
+        if ($LASTEXITCODE -ne 0) {
+            throw "$LASTEXITCODE Pester tests failed"
```
Comment (JamesWTruher) [inline build.psm1]: not sure this is needed anymore, since `Test-PSPesterResults` now throws

File: tools/appveyor.psm1 [hunk 3]
```diff
@@ -96,34 +96,26 @@ function Invoke-AppVeyorTest
...
-
-    Write-Host -Foreground Green 'Upload FullCLR test results'
-    $x = [xml](cat -raw $testResultsFileFullCLR)
+    Test-PSPesterResults -TestResultsFile $testResultsFile
 
-    if ([int]$x.'test-results'.failures -gt 0)
-    {
-        throw "$($x.'test-results'.failures) tests in test/fullCLR failed"
-    }
+    Test-PSPesterResults -TestResultsFile $testResultsFileFullCLR
```
Comment (JamesWTruher) [inline tools/appveyor.psm1]: hmmm - a failure in reading $testResultsFile may keep this from running

File: build.psm1 (hunk 1 above)
Comment (TravisEz13) [inline build.psm1]: I'm not sure if this is needed any more... unrestricted is the default... I just didn't want to change it.

File: build.psm1 (hunk 2 above)
Comment (TravisEz13) [inline build.psm1]: Agreed, I'll remove it.

**Comment by JamesWTruher** [comment]
:shipit:

**Pull request merged**